a3fe/tests/test_run.py

```diff
@@ -332,7 +335,7 @@ def _mock_run_process_inner(
 
     @staticmethod
     @pytest.fixture(scope="class")
-    def setup_calc(mock_run_process):
+    def setup_calc(mock_run_process, system_prep_config):
```

Comment:
To make this generic, I think you'll need to pass in `engine_type`, which should then be supplied to `a3.Calculation` - otherwise e.g. a GROMACS system preparation config will break things.